web/src/components/profile/ProfileAddress.test.tsx

```diff
@@ -0,0 +1,824 @@
+    render(
+      <ThemeProvider theme={createTheme()}>
+        <WithStatefulAddressData
+          initialData={emptyAddressData}
+          municipalities={municipalities || [generateMunicipality({})]}
+        >
+          <ProfileAddress businessLocationType="NJ" />
+        </WithStatefulAddressData>
+      </ThemeProvider>
+    );
+  };
+
+  const renderNonNjUSBusinessComponent = (): void => {
+    render(
+      <ThemeProvider theme={createTheme()}>
+        <WithStatefulAddressData initialData={emptyAddressData} municipalities={[generateMunicipality({})]}>
+          <ProfileAddress businessLocationType="US" />
+        </WithStatefulAddressData>
+      </ThemeProvider>
+    );
+  };
+
+  const renderIntlBusinessComponent = (): void => {
+    render(
+      <ThemeProvider theme={createTheme()}>
+        <WithStatefulAddressData initialData={emptyAddressData} municipalities={[generateMunicipality({})]}>
+          <ProfileAddress businessLocationType="INTL" />
+        </WithStatefulAddressData>
+      </ThemeProvider>
+    );
+  };
+
+  describe("NJ business location", () => {
+    it("renders profile address line 1 field with input value", () => {
+      const address = generateAddress({
+        addressLine1: "1111 Home Alone",
+      });
+      useMockBusiness(
+        generateBusiness({
+          formationData: generateFormationData({
+            completedFilingPayment: false,
+            formationFormData: generateFormationFormData({ ...address }),
+          }),
+        })
+      );
+      renderNjBusinessComponent({});
+      expect(screen.getByLabelText("Address line1")).toHaveValue("1111 Home Alone");
+    });
+
+    it("renders profile address line 1 reflects changed value", () => {
+      const address = generateAddress({
+        addressLine1: "",
+      });
+      useMockBusiness(
+        generateBusiness({
+          formationData: generateFormationData({
+            completedFilingPayment: false,
+            formationFormData: generateFormationFormData({ ...address }),
+          }),
+        })
+      );
+      renderNjBusinessComponent({});
+
+      expect(screen.getByLabelText("Address line1")).toHaveValue("");
+      fillText("Address line1", "1111 Home Alone");
+      expect(screen.getByLabelText("Address line1")).toHaveValue("1111 Home Alone");
+    });
+
+    it("renders profile address line 1 error", () => {
+      const address = generateAddress({
+        addressLine1: "",
+      });
+      useMockBusiness(
+        generateBusiness({
+          formationData: generateFormationData({
+            completedFilingPayment: false,
+            formationFormData: generateFormationFormData({ ...address }),
+          }),
+        })
+      );
+      renderNjBusinessComponent({});
+
+      blurInput("Address line1");
+      expect(screen.getByText(Config.formation.fields.addressLine1.error)).toBeInTheDocument();
+    });
+
+    it("renders profile address line 2 field with input value", () => {
```

Comment:
[sand]: the tests for address line 2 could benefit from similar updates to what was in my line 1 suggestions